### PR TITLE
Add thread_executors module

### PR DESCRIPTION
**Pull request by aurianer**
Move thread executors into a module

### CODE REVIEW

**Comment by msimberg** [comment]
For these kinds of failures https://cdash.cscs.ch//viewBuildError.php?buildid=101308, you can either directly include the correct executor header, but you can also add the collective thread executors headers to `hpx/include/parallel_executors.hpp` (and vice versa) for compatibility.

File: libs/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2019-2020 The STE||AR-Group
+# Copyright (c) 2020 The STE||AR-Group
```
Comment (msimberg) [inline libs/CMakeLists.txt:1]: Not a big deal, but we can probably revert this change and have `2019-2020` in `create_library_skeleton.py`.

File: hpx/runtime/threads/executors.hpp [hunk 2]
```diff
@@ -10,11 +10,11 @@
...
-#include <hpx/runtime/threads/executors/current_executor.hpp>
-#include <hpx/runtime/threads/executors/default_executor.hpp>
-#include <hpx/runtime/threads/executors/service_executors.hpp>
-#include <hpx/runtime/threads/executors/this_thread_executors.hpp>
-#include <hpx/runtime/threads/executors/embedded_thread_pool_executors.hpp>
+#include <hpx/thread_executors/current_executor.hpp>
+#include <hpx/thread_executors/default_executor.hpp>
+#include <hpx/thread_executors/service_executors.hpp>
+#include <hpx/thread_executors/this_thread_executors.hpp>
+#include <hpx/thread_executors/embedded_thread_pool_executors.hpp>
```
Comment (msimberg) [inline hpx/runtime/threads/executors.hpp:17]: I think this whole file (`hpx/runtime/threads/executors.hpp`) can move as well, but it can be replaced by the generated module header. You can then point the compatibility header to `hpx/thread_executors.hpp` instead.

File: hpx/runtime/threads/executors.hpp [hunk 3]
```diff
@@ -10,11 +10,11 @@
...
 #include <hpx/runtime/threads/thread_executor.hpp>
```
Comment (msimberg) [inline hpx/runtime/threads/executors.hpp:18]: Can this file (`hpx/runtime/threads/thread_executor.hpp`) be moved as well or does it cause problems? `limiting_executor` and `sequenced_executor` don't actually need it. `thread_execution_information.hpp` and `thread_execution.hpp` probably need it. If they need we can probably move the file to the `execution` module at least. It already contains some of the base functionality of `thread_executors` and will be removed soon anyway.

File: hpx/runtime/threads/executors.hpp (hunk 2 above)
Comment (aurianer) [reply hpx/runtime/threads/executors.hpp:17]: Yep!

File: hpx/runtime/threads/executors.hpp (hunk 3 above)
Comment (aurianer) [reply hpx/runtime/threads/executors.hpp:18]: Ok!

Comment (msimberg) [reply hpx/runtime/threads/executors.hpp]: Let's keep `hpx/runtime/threads/thread_executor.hpp` where it was: https://176179-4455628-gh.circle-artifacts.com/0/hpx/artifacts/circular_deps.svg. This is not a dependency worth spending time on.

**Comment by aurianer** [comment]
solved :)
